playkit/src/main/java/com/kaltura/playkit/player/TrackSelectionHelper.java

```diff
@@ -617,13 +617,12 @@ private boolean isAdaptive(int rendererIndex, int groupIndex) {
      * Validate and return parsed uniqueId.
      *
      * @param uniqueId - uniqueId to validate
-     * @return - parsed uniqueId in case of success.
-     * @throws IllegalArgumentException when uniqueId is illegal.
+     * @return IllegalArgumentException when uniqueId is illegal ele return null
      */
-    private int[] validateUniqueId(String uniqueId) throws IllegalArgumentException {
+    public IllegalArgumentException validateUniqueId(String uniqueId) {
```

Comment:
Returning an exception from a function looks weird. Maybe instead, throw the message like before but be prepared to catch it in ExoPlayerWrapper.
So, no change in TrackSelectionHelper (make it like it was before this commit) and in ExoPlayerWrapper add `try..catch (IllegalArgumentException)` around changeTrack().

Reply:
done